### ISSUE TITLE
linter-eslint appears to delete .eslintcache when an error is found

### ISSUE THREAD

**Issue by dominicbarnes** [AtomLinter/linter-eslint#587]
I'm running ESLint with the `--cache` flag in my repo from my `package.json`, doing so generates a `.eslintcache` file at the root of my project.

When a `.eslintcache` file exists, and I make a change to my file that introduces a lint error, (such as removing a semicolon) it deletes the `.eslintcache` file around the time that it displays the lint error in the editor. (which I didn't expect)

I only get this behavior from linter-eslint, not from ESLint itself. To demonstrate this, I ran ESLint and generated the cache file. Using another editor I introduced a lint error, then ran ESLint again, and the file was not deleted. If I disable `linter-eslint`, the behavior goes away. (I've also tried tweaking several configs to see if they change this behavior, to no avail)

I looked through the source code and didn't find any mention of `.eslintcache`, so I'm not entirely sure how this behavior is coming about.

In short, it would be great if the `.eslintcache` file was not deleted by linter-eslint. (or maybe it's linter, as I said it's not super-clear where the logic for deleting this file is coming from)

**Comment by Arcanemagus** [comment]
@IanVS Should this be filed over on the ESLint repo? Nothing here deals with that, so it must be ESLint that is deleting that when it isn't specified as an option.

**Comment by dominicbarnes** [comment]
@Arcanemagus looks like you are right, if I run `eslint` without the `--cache` option, it deletes the cache file. I don't know if this expected behavior or not, so I'll investigate what the expected behavior is from ESLint. (it seems peculiar that it would delete a previous cache file just because you didn't use the flag)

Are there any plans to respect the `.eslintcache` from this project?

**Comment by Arcanemagus** [comment]
I'm not sure how that option is really relevant to this project... for starters by definition this is only ran when a file is modified. Secondly... this only executes `eslint` on raw text, never on a file itself.

I'm going to mark this as closed since it's an external issue with `eslint` itself that really can't be fixed here.

**Comment by IanVS** [comment]
I'm going to look into this.  It seems to me that executing on text as we
are should not involve the cache at all.  But yes, this should likely be
fixed upstream in ESLint if that's truly what's happening.

**Comment by dominicbarnes** [comment]
@IanVS it turns out this is actually expected behavior by ESLint, if you do not specify `cache: true` (and you don't have a custom `cacheFile`) it will remove the `.eslintcache`, since it considers running again w/o a cache to invalidate any previous cache. (and it doesn't look like they're keen on changing that behavior)

I think the best solution here would be to add a config to turn on caching support, so you can at least add `cache: true` if you see a `.eslintcache` file in the project root. (not sure if that has other implications though, since you're running on the text of the file which will likely not be what's already saved on disk)

**Comment by IanVS** [comment]
Yes, that's expected when you're linting files, but I think that in our case with linter-eslint where we are linting straight text, ESLint should not invalidate the cache.  Did you open an issue at ESLint?  I'll jump in and talk with the rest of the ESLint team about it.

**Comment by dominicbarnes** [comment]
@IanVS I didn't open my own ticket, but I did chime in on eslint/eslint#4943 ([comment](https://github.com/eslint/eslint/issues/4943#issuecomment-220875279))

**Comment by IanVS** [comment]
FYI @dominicbarnes, this should now be fixed (cache should no longer be deleted) in version `3.0.0` of ESLint.
